### PR TITLE
Occt variants

### PR DESCRIPTION
**Pull request by looooo**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by looooo** [comment]
@adrianinsaval sorry for pushing to FreeCAD/freecad-feedstock . Somehow my fork was directing to this repository. Feel free to delete the branch once this PR is merged.

**Comment by looooo** [comment]
no idea why some builds are failing. Something about pyside2 packages:
```
KeyError: Dist(channel='conda-forge', dist_name='pyside2-5.15.8-py310h1570d6a_3', name='pyside2', fmt='.conda', version='5.15.8', build_string='py310h1570d6a_3', build_number=3, base_url=None, platform=None)
```
any ideas what that can be @conda-forge/pyside2

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 MACOSX_DEPLOYMENT_TARGET:  # [osx]
 - '11.0'                   # [osx and arm64]
-- '10.13'                  # [osx and x86_64]
\ No newline at end of file
+- '10.13'                  # [osx and x86_64]
+occt:
+  - 7.7.2
+  - 7.6.3
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:6]: what about just unpinning?

Comment (looooo) [reply recipe/conda_build_config.yaml:6]: we wanted to build against 7.6.3 and 7.7.2. There are some problems reported with 7.7.2 that's why we still want to keep 7.6.3

**Comment by adrianinsaval** [comment]
tbh I think we can just keep it with occt 7.6.3 only until next release, I hope we can fix 7.7 problems by then.

> @adrianinsaval sorry for pushing to FreeCAD/freecad-feedstock . Somehow my fork was directing to this repository. Feel free to delete the branch once this PR is merged.

that probably happened because that used to be your personal fork that then got transferred to the org account. Right now https://github.com/looooo/freecad-feedstock redirects to https://github.com/FreeCAD/freecad-feedstock, I guess you need to fork again for the redirect to stop

**Comment by looooo** [comment]
> tbh I think we can just keep it with occt 7.6.3 only until next release, I hope we can fix 7.7 problems by then. I don't oppose merging this though, so if you think it's worth doing go ahead and merge! I've been facing those weird KeyError problems many times lately, it seems to happen at random, running CI again on the exact same code may succeed at a different time.

In my mind we should align with Conda-forge-pinning which currently is at occt 7.7. But we can keep occt 7.6.3 if needed, (maybe supporting only some python versions to make the build-matrix smaller)

**Comment by adrianinsaval** [comment]
maybe we can just move conda-forge to occt 7.7, for people who encounter occt problems they can move to the freecad channel where I'm going to keep occt 7.6.3+py 3.10 builds

**Pull request merged**